packages/app-cognito-authenticator/src/hooks/useSignIn.ts

```diff
@@ -21,6 +22,11 @@ interface Reducer {
     (prev: State, next: Partial<State>): State;
 }
 
+type CheckContactParams = {
+    verified: Record<any, any>;
```

Comment:
Please figure out better types.

Reply:
@brunozoric improved `Record<any, any>` type

Reply:
@neatbyte-ivelychko putting `object` to `AuthData` interface instead of Record<any, any> is same bad type written in another way.
The `verified` is not used anywhere except two checks if it actually is defined, so maybe try to figure out what is contained in it? Don't spend much more time on it. If you can't find the correct type, put
```
{verified?: Record<string, string | number | boolean>}
```
for now.

Also, I see that unverified is not used anywhere, so maybe it is not necesary at the moment...?

Reply:
But still we need to add `object` in the `AuthData` type `[key: string]: string | null | boolean | undefined | object;` property. Becuase TS keeps arguing that that type does not support object, it only supports `string | null | boolean | undefined`.

Reply:
When logging the data variable, I get:
```
{
    "verified": {
        "email": "my@email.com"
    },
    "unverified": {}
}
```

Maybe it could be like this?
```
export interface AuthDataVerified {
    email?: string;
}

export interface AuthDataUnverified {
    someUnknownKey?: string;
}

export interface AuthData {
    username?: string;
    email?: string;
    verified?: AuthDataVerified;
    unverified?: AuthDataUnverified;
    [key: string]: string | null | boolean | undefined | AuthDataVerified | AuthDataUnverified;
}
```

Reply:
Okay, I have also checked method `verifiedContact` in `@aws-amplify` and I saw that besides `email` property they have `phone_number` property. Should we consider adding it to the `verified` and `unverified`.

```
if (attrs['email']) {
   if (attrs['email_verified']) {
	verified['email'] = attrs['email'];
   } else {
	unverified['email'] = attrs['email'];
   }
}
 if (attrs['phone_number']) {
   if (attrs['phone_number_verified']) {
	verified['phone_number'] = attrs['phone_number'];
    } else {
      unverified['phone_number'] = attrs['phone_number'];
    }
}
```

Reply:
@neatbyte-ivelychko yes, add anything you can find. Maybe we don't use those vars now, but let's have them in types.

Reply:
Added `phone_number` to the type

Reply:
@brunozoric I think that this might be the latest batch. Because after I have fixed type for [ComposableFC](https://github.com/webiny/webiny-js/pull/3774/commits/bafd2c426b526502c33f0109c7171abe7694179e#diff-9b4cf473a633d9d81cced171e933afa44933883f61041bb0c9cdac69853ed6a5R5) project had been built without any errors.

We just need to check whether it is gonna pass CI/CD.

Reply:
Nevermind, it was the issue with my local build. Somehow it was always green event if build had errors